pants_plugins/sendwave/pants_docker/test_sources.py

```diff
@@ -16,9 +15,8 @@
 @pytest.fixture
 def rule_runner() -> RuleRunner:
     rule_runner = RuleRunner(
-        target_types=[PythonLibrary, PythonTests, Files, Resources, RelocatedFiles],
+        target_types=[PythonSourceTarget, FileTarget, ResourceTarget],
         rules=[
-            *pytest_runner.rules(),
```

Comment:
I kept getting obscure rule graph errors whenever I included `pytest_runner.rules()` ☹️ I made sure everything was set up [according to the docs](https://www.pantsbuild.org/docs/rules-api-testing#setting-up-the-rulerunner) by adding `QueryRule(TestResult, [PythonTestFieldSet])` but this bumped the number of errors from 3 to 24! My suspicion is that there's an additional set of rules that must be included in order to provide the dependencies for `pytest_runner.rules()` but I couldn't figure out what they were 😕 

Brute force process-of-elimination debugging revealed that this line was causing the rule graph errors, and if I removed it all of the tests passed. I'd love to figure out how to get this working (and what it's useful for) but in the meantime do you feel comfortable removing this as long as the tests continue to pass?

Reply:
yeah I looked into that a bit and it seemed super weird - I also had a lot of trouble getting it set up originally (asking on the pants slack, has in the past, been super useful (and one of the maintainers was interested in helping me get set up previously too! so that was neat)) 

But i was able to get it to work by removing this and including the PythonGeneratorSourcesFile so I pushed that up. I think uh... as long as it's running (and it seems like it is!) then we're ok! but this could probably be more tested than it is